lib/sdp.c

```diff
@@ -1997,8 +2001,11 @@ int sdp_set_uuidseq_attr(sdp_record_t *rec, uint16_t aid, sdp_list_t *seq)
 	}
 	if (status == 0) {
 		sdp_data_t *data = sdp_seq_alloc(dtds, values, len);
-		sdp_attr_replace(rec, aid, data);
-		sdp_pattern_add_uuidseq(rec, seq);
+		if (data != NULL) {
+			sdp_attr_replace(rec, aid, data);
+			sdp_pattern_add_uuidseq(rec, seq);
+		} else
+			return -ENOMEM;
```

Comment:
supposed to return -1 and set errno  = ENOMEM or return ENOMEM itself?
Same comment applies to all return statements introduced in this pull request